README.md

```diff
@@ -256,61 +255,69 @@ func TestServerCompatabilityWithOpenAPI(t *testing.T) {
 
 > **Tip**
 > 
-> Try changing the spec in a way that is incompatible with the API and run the server again.
+> Try changing the spec in a way that is incompatible with the API and run the 
+> server again.
 > 
-> Cellotape is reporting to you the incompatibility and allow you to make sure your implementation and spec are in sync. 
+> Cellotape reports any incompatibility, enabling you to check that your 
+> implementation and spec are in sync. 
 
 ## Loading OpenAPI spec (`router.OpenAPISpec`)
 
-To init a new Cellotape HTTP router you must first load an OpenAPI spec.
-
-The OpenAPI spec defines URL paths and HTTP methods for different operations.
-
-Cellotape routes HTTP call on these paths and methods for the relevant handler implementation of each operation.
+To initialize a new Cellotape HTTP router you must first load an OpenAPI spec. 
 
-Cellotape also validates the handler is implementing correctly the request and response defined in the spec.
+The OpenAPI spec defines URL paths and HTTP methods for various operations. 
+Cellotape routes HTTP calls on these paths and methods for the relevant handler 
+implementation of each operation. Cellotape also validates that the handler 
+correctly implements the request and response defined in the spec.
 
-There are few ways you can load the OpenAPI spec to be used with Cellotape.
+There are several ways you can load the OpenAPI spec for use with Cellotape.
 
 ### Using FS embedding (recommended)
 
-Using [Go embedding](https://pkg.go.dev/embed) you can embed your OpenAPI YAML or JSON file directly with the compiled binary of your app. 
+Using [Go embedding](https://pkg.go.dev/embed), you embed the OpenAPI YAML or 
+JSON file with the compiled binary of your app. For example, embedding an 
+`openapi.yaml` like this:
 
-For example embedding an `openapi.yaml` as follows. 
 ```go
 //go:embed openapi.yaml
 var specData []byte
 ```
 
-This allows you to init then the spec from its bytes with `router.NewSpecFromData` as follows:
+This enables you to initialize the spec from its bytes with `router.NewSpecFromData` 
+like this:
 
 ```go
 spec, err := router.NewSpecFromData(specData)
 ```
 
-`router.NewSpecFromData` returns the loaded spec object and an error if failed to parse and validate it.
+`router.NewSpecFromData` returns the loaded spec object or an error if the object 
+fails to parse and validate.
```

Comment:
I used **and** because Go functions can return multiple values.
In this case:
```go
spec, err := router.NewSpecFromData(specData)
```
The function returns `spec` **and** an `err` (error).
The error is `nil` (empty) if the spec parsed and validated correctly but it is still a `nil` value returned from the function for the err value.

Please let me know if you still think it should be **or**.
Notice this is a very common pattern in Go.

Reply:
Thanks, understood, reverted

Reply:
Technically it returns both at the same time but with one of them having a "zero value" or `nil`.
Few examples from [effective go](https://go.dev/doc/effective_go):
- https://go.dev/doc/effective_go#:~:text=return%20a%20count%20and%20an%20error
- https://go.dev/doc/effective_go#:~:text=The%20method%20returns%20the%20number%20of%20bytes%20read%20and%20an%20error%20value
- https://go.dev/doc/effective_go#:~:text=returns%20a%20value%20and%20an%20error